pkg/scheduler/server/server.go

```diff
@@ -236,6 +238,7 @@ func (s *Server) runEtcd(ctx context.Context) error {
 		return err
 	case <-ctx.Done():
 		log.Info("Embedded Etcd shutting down")
+		cron.Wait()
```

Comment:
We should also be waiting when etcd returns an error.

Reply:
Let me see if I can add that. I am just changing the cron lib use to match the latest version.

Reply:
There is a defer on etcd close() already.